### PR TITLE
20_2 TreeList Cell Editing: rewrite

### PR DESCRIPTION
**Pull request by pavel-2020**
The text is based on https://github.com/DevExpress/devextreme-demos/blob/20842f6a158b112ac83d9df67cdf0b542d6a781f/JSDemos/Demos/DataGrid/CellEditingAndEditingAPI/description.md
https://github.com/DevExpress/devextreme-demos/pull/783

### CODE REVIEW

File: JSDemos/Demos/TreeList/CellEditing/description.md [hunk 1]
```diff
@@ -1 +1,7 @@
...
\ No newline at end of file
+Users can modify TreeList data cell by cell. In this mode, only one cell can be in the edit state at a time. TreeList saves changes immediately after the focus leaves cell.
+
+To enable the cell mode, do the following:
+- Set the **editing**.[mode](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#mode) to *"cell*".
+- Assign **true** to the [editing](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/) object's [allowUpdating](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowUpdating), [allowAdding](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowAdding), and [allowDeleting](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowDeleting) properties.
+
+When a user presses the button to add a new row, the [onInitNewRow](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#onInitNewRow) handler populates the new row editor with data. In this demo, this function preselects *Not Started* in the `Status` drop-down field and presets the current date for the `Start Date` and `Due Date` fields.
```
Comment (AlexJameson) [inline JSDemos/Demos/TreeList/CellEditing/description.md]: > this function preselects *Not Started* in the `Status` drop-down field

This is not a drop-down field. The drop-down field is on the screenshot: 
image

Also I still don't get what is a 'row editor'

File: JSDemos/Demos/TreeList/CellEditing/description.md [hunk 2]
```diff
@@ -1 +1,7 @@
...
\ No newline at end of file
+Users can modify TreeList data cell by cell. In this mode, only one cell can be in the edit state at a time. TreeList saves changes immediately after the focus leaves cell.
+
+To enable the cell mode, do the following:
+- Set the **editing**.[mode](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#mode) to *"cell*".
+- Assign **true** to the [editing](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/) object's [allowUpdating](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowUpdating), [allowAdding](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowAdding), and [allowDeleting](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowDeleting) properties.
+
+When a user presses the button to add a new row, the [onInitNewRow](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#onInitNewRow) handler populates cells of a new row with default values. In this demo, this function sets *Not Started* as Status and current date as Start Date and Due Date.
```
Comment (AlexJameson) [inline JSDemos/Demos/TreeList/CellEditing/description.md]: I feel like some formatting is required for the mentioned fields:

```suggestion
When a user presses the button to add a new row, the [onInitNewRow](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#onInitNewRow) handler populates cells of a new row with default values. In this demo, this function sets *Not Started* as `Status` and current date as `Start Date` and `Due Date`.
```

**Review by AlexJameson — DISMISSED** [review]

File: JSDemos/Demos/TreeList/CellEditing/description.md [hunk 3]
```diff
@@ -1 +1,7 @@
...
\ No newline at end of file
+Users can modify TreeList data cell by cell. In this mode, only one cell can be in the edit state at a time. TreeList saves changes immediately after the focus leaves cell.
+
+To enable the cell mode, do the following:
+- Set the **editing**.[mode](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#mode) to *"cell*".
+- Assign **true** to the [editing](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/) object's [allowUpdating](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowUpdating), [allowAdding](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowAdding), and [allowDeleting](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowDeleting) properties.
+
+When a user presses the button to add a new row, the [onInitNewRow](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#onInitNewRow) handler populates cells of a new row with default values. In this demo, this function sets *Not Started* as `Status` and current date as `Start Date` and `Due Date`.
```
Comment (RomanTsukanov) [inline JSDemos/Demos/TreeList/CellEditing/description.md]:
```suggestion
This demo also shows how to populate cells of a new row with default values. The values are assigned within the [onInitNewRow](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#onInitNewRow) event handler.
```

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

**Review by RomanTsukanov — DISMISSED** [review]

**Review by AlexJameson — DISMISSED** [review]

File: JSDemos/Demos/TreeList/CellEditing/description.md [hunk 4]
```diff
@@ -1 +1,7 @@
-This example demonstrates editing in _&laquo;cell&raquo;_ mode. A&nbsp;click on&nbsp;any cell switches it&nbsp;into the editing state. Once you click outside the cell, the TreeList saves your changes. Note that several columns are made uneditable by&nbsp;setting their **allowEditing** property to _false_.
\ No newline at end of file
+Users can modify TreeList data cell by cell. In this mode, only one cell can be in the edit state at a time. TreeList saves changes immediately after the focus leaves cell.
```
Comment (arminal) [inline JSDemos/Demos/TreeList/CellEditing/description.md]:
```suggestion
Users can modify TreeList data cell by cell. In this mode, only one cell can be in the edit state at a time. The TreeList saves changes immediately after the focus leaves cell.
```
the focus leaves cell? rephrase

File: JSDemos/Demos/TreeList/CellEditing/description.md [hunk 5]
```diff
@@ -1 +1,7 @@
...
+
+To enable the cell mode, do the following:
```
Comment (arminal) [inline JSDemos/Demos/TreeList/CellEditing/description.md]:
```suggestion
To enable cell mode, do the following:
```

File: JSDemos/Demos/TreeList/CellEditing/description.md [hunk 6]
```diff
@@ -1 +1,7 @@
...
+- Set the **editing**.[mode](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#mode) to *"cell*".
```
Comment (arminal) [inline JSDemos/Demos/TreeList/CellEditing/description.md]:
```suggestion
- Set  **editing** [mode](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#mode) to *"cell*".
```

File: JSDemos/Demos/TreeList/CellEditing/description.md [hunk 7]
```diff
@@ -1 +1,7 @@
-This example demonstrates editing in _&laquo;cell&raquo;_ mode. A&nbsp;click on&nbsp;any cell switches it&nbsp;into the editing state. Once you click outside the cell, the TreeList saves your changes. Note that several columns are made uneditable by&nbsp;setting their **allowEditing** property to _false_.
\ No newline at end of file
+Users can modify TreeList data cell by cell. In this mode, only one cell can be in the edit state at a time. The TreeList saves changes immediately after the focus leaves cell.
+
+To enable cell mode, do the following:
+- Set  **editing** [mode](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#mode) to *"cell*".
```
Comment (RomanTsukanov) [inline JSDemos/Demos/TreeList/CellEditing/description.md]:
```suggestion
- Set the **editing**.[mode](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#mode) property to *"cell*".
```

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

**Review by RomanTsukanov — APPROVED** [review]

**Pull request merged**